external/fv3fit/fv3fit/_shared/jacobian.py

```diff
@@ -73,3 +72,56 @@ def compute_standardized_jacobians(model, data, input_variables):
     jacobians = get_jacobians(model, avg_profiles)
     std_jacobians = standardize_jacobians(jacobians, data)
     return std_jacobians
+
+
+def standardize_jacobians_by_vertical_level(
+    all_jacobians: Mapping[str, OutputSensitivity],
+    sample: Mapping[str, tf.Tensor],
+    units: str = "dimensionless",
+) -> Mapping[str, OutputSensitivity]:
+    """
+    Generate sensitivity jacobians for each output of a model and
+    standardize (dimensionless or output units) for easy
+    inter-variable comparison.
+
+    The scaling uses the standard deviation of each vertical level
+    for the input (std_input) and output
+    (std_output) sample. The associated
+    jacobian result is multiplied by [ std_input / std_output ].
+    If units are "output" the associated jacobian result is multiplied
+    by [ std_input ].
+    """
+
+    # normalize factors so sensitivities are comparable but still
+    # preserve level-relative magnitudes
+    std_factors = {name: data.std(axis=0) for name, data in sample.items()}
+    standardized_jacobians: Dict[str, OutputSensitivity] = {}
+    for out_name, per_input_jacobians in all_jacobians.items():
```

Comment:
I think this for loop would also work for `standardize_jacobian`. Can you refactor it into two shared helpers and remove the `units` flag.
-  `units='output'`. `jacobian_to_output_change(all_jacobians, delta_input)`
- `units='dimensionless'`. `nondimensionalize_jacobian(all_jacobian, scales: Mapping[str, np.ndarray])`

At this point the standard deviation code is a one liner...I'd delete `standardize_jacobians_by_vertical_level` and `standardize_jacobians` and replace any of their uses with
```
# andre's scaling
std_factors = {name: data.std() for name, data in sample.items()}
# your scaling
# std_factors = {name: data.std(axis=0) for name, data in sample.items()}
nondimensionalize_jacobian(all_jacobian, std_factors)
```
this will easily allow downstream visualizations to change the scale factors without adding another function to this module.